test/index.js

```diff
@@ -84,49 +84,65 @@ test('returns 0', t => {
   t.deepEqual(style, { width: 0 })
 })
 
-test('returns an array of responsive style objects', t => {
+test('returns merged object of responsive styles', t => {
   const style = width({
     width: ['100%', '50%'],
   })
-  t.deepEqual(style, [
-    { width: '100%' },
-    { '@media screen and (min-width: 40em)': { width: '50%' } },
-  ])
+  t.deepEqual(style, {
+    width: '100%',
+    '@media screen and (min-width: 40em)': { width: '50%' },
+  })
 })
 
-test('returns an array of responsive style objects for all breakpoints', t => {
+test('returns merged object of responsive styles for all breakpoints', t => {
   const style = width({
     width: ['100%', '75%', '50%', '33%', '25%'],
   })
-  t.deepEqual(style, [
-    { width: '100%' },
-    { '@media screen and (min-width: 40em)': { width: '75%' } },
-    { '@media screen and (min-width: 52em)': { width: '50%' } },
-    { '@media screen and (min-width: 64em)': { width: '33%' } },
-  ])
+  t.deepEqual(style, {
+    width: '100%',
+    '@media screen and (min-width: 40em)': { width: '75%' },
+    '@media screen and (min-width: 52em)': { width: '50%' },
+    '@media screen and (min-width: 64em)': { width: '33%' },
+  })
 })
 
 test('skips undefined responsive values', t => {
   const style = width({
     width: ['100%', , '50%'],
   })
-  t.deepEqual(style, [
-    { width: '100%' },
-    { '@media screen and (min-width: 52em)': { width: '50%' } },
-  ])
+  t.deepEqual(style, {
+    width: '100%',
+    '@media screen and (min-width: 52em)': { width: '50%' },
+  })
 })
 
+// Previous object parsing test
+// test('parses object values', t => {
+//   const style = width({
+//     width: {
+//       _: '100%',
+//       2: '50%',
+//     },
+//   })
+//   t.deepEqual(style, {
+//     width: '100%',
+//     '@media screen and (min-width: 64em)': { width: '50%' },
+//   })
+// })
+
+// Object parsing work as expected...
+// Previous behaviour was kind of buggy for me
 test('parses object values', t => {
   const style = width({
     width: {
-      _: '100%',
-      2: '50%',
+      0: '100%',
+      1: '50%',
```

Comment:
The test above was for showing how breakpoints could be "skipped" – why did you change this?

Reply:
In your previous example it was not skipped, so there was a bug.

https://github.com/styled-system/styled-system/blob/master/test/index.js#L128

It is now fixed, if the key does not point to anything it will be skipped.